### PR TITLE
Support RISC-V p-ext-proposal v0.9.2

### PR DESCRIPTION
**Pull request by hope51607**
See the previous PR #572 

Hello, we’re the team from Programming Langauge Lab, NTHU (National Tsing-Hua University), Taiwan and also from Peakhills group.

This PR implements all instructions in the P-Extension Proposal v0.9.2 ([link](https://github.com/riscv/riscv-p-spec/blob/master/P-ext-proposal.adoc)).

I also create riscv/riscv-opcodes#60 PR to add p-ext opcodes in riscv-opcodes.

### CODE REVIEW

**Comment by aswaterman** [comment]
Thank you (again!) for the contribution. We’ll wait for the RVIA zeitgeist to clarify, but hopefully we can merge these patches soon.

**Comment by aswaterman** [comment]
Now that I've looked at the patches, I still think we need to wait until the major opcode issue is resolved.  `1111111` is reserved for >32-bit instructions, so shouldn't be used by the P extension.  This will get fixed soon through RVIA, I expect.

**Comment by hope51607** [comment]
@aswaterman
I have updated encoding to p-ext v0.9.2 which change major opcode "GE80B 1111111" to "OP-P 1110111".

File: README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Spike supports the following RISC-V ISA features:
   - B extension, v0.92
   - K extension, v0.8.1 ([Scalar Cryptography](https://github.com/riscv/riscv-crypto))
   - V extension, v0.9, w/ Zvlsseg/Zvamo/Zvqmac, w/o Zvediv, (_requires a 64-bit host_)
+  - P extension, v0.9.1
```
Comment (kito-cheng) [inline README.md]: @hope51607 Could you update version info here? I think it should be 0.9.2 according your latest subject of this PR.

**Comment by kito-cheng** [comment]
And seems like you need to rebase due to the conflict.

**Comment by hope51607** [comment]
@kito-cheng 

Thanks for the review.
I have updated `readme.md` and rebased master branch.

**Comment by kito-cheng** [comment]
@aswaterman the major opcode is updated, does here any issue for this PR need to fix?

**Comment by aswaterman** [comment]
I'd like to wait for RVIA to signal that the opcode allocation is acceptable, then I'm happy to merge.

**Comment by kito-cheng** [comment]
Oh, I didn't aware it's not accept by RVIA, I thought P spec has updated to using that means already accepted, but obviously I am wrong.

**Comment by chuanhua** [comment]
@aswaterman Krste and I have discussed this and agreed that the major opcode 1110111 (named OP-P in the P extension specification) is a good location for the P extension to do further development work before ratification. Could you help to merge this pull request? Thanks!

**Comment by aswaterman** [comment]
Great, thanks for the note, @chuanhua.

@hope51607 is this version of the PR ready to merge?

**Comment by hope51607** [comment]
@aswaterman yes, this is the version ready to merge, thanks.

**Comment by aswaterman** [comment]
Cool.  I'll give it another code review this weekend.  It'll get merged in soon.

**Comment by aswaterman** [comment]
@chihminchao can I recruit your assistance in [re-]reviewing this patch set?

**Comment by chihminchao** [comment]
@aswaterman of course,  I will review this weekend.

**Review by chihminchao — CHANGES_REQUESTED** [review]
There is an lost tab.  line 1323

**Comment by chihminchao** [comment]
@hope51607  a lost tab is still in disasm.cc. Please help to correct it.

@aswaterman  All parts looks good to me

**Comment by hope51607** [comment]
> @hope51607 a lost tab is still in disasm.cc. Please help to correct it.

Done.

**Pull request merged**